pyproject.toml

```diff
@@ -1,11 +1,140 @@
+[tool.pytest.ini_options]
+addopts = "-q"
+filterwarnings = [
+    "error", # Fail the tests if there are any warnings.
+    "ignore:^find_module\\(\\) is deprecated and slated for removal in Python 3.12; use find_spec\\(\\) instead$:DeprecationWarning:importlib",
+    "ignore:^FileFinder.find_loader\\(\\) is deprecated and slated for removal in Python 3.12; use find_spec\\(\\) instead$:DeprecationWarning:importlib",
+    "ignore:^pkg_resources is deprecated as an API:DeprecationWarning:pkg_resources",
+    "ignore:^Deprecated call to .pkg_resources\\.declare_namespace\\('.*'\\).\\.:DeprecationWarning:pkg_resources",
+]
+
+[tool.pydocstyle]
+ignore = [
+    # Missing docstrings.
+    "D100","D101","D102","D103","D104","D105","D106","D107",
+
+    # "No blank lines allowed after function docstring" conflicts with the
+    # Black code formatter which insists on inserting blank lines after
+    # function docstrings.
+    "D202",
+
+    # "1 blank line required before class docstring" conflicts with another
+    # pydocstyle rule D211 "No blank lines allowed before class docstring".
+    "D203",
+
+    # "Multi-line docstring summary should start at the first line"
+    # and "Multi-line docstring summary should start at the second line".
+    # These two rules conflict with each other so you have to disable one of them.
+    # How about we disable them both? PEP 257 says either approach is okay:
+    #
+    # > The summary line may be on the same line as the opening quotes or on
+    # > the next line.
+    # >
+    # > https://peps.python.org/pep-0257/#multi-line-docstrings
+    "D212",
+    "D213",
+]
+
+[tool.coverage.run]
+branch = true
+parallel = true
+source = ["lms", "tests/unit"]
+omit = [
+    "*/lms/__main__.py",
+    "lms/pshell.py",
+    "lms/migrations/*",
+    "lms/extensions/feature_flags/views/test.py",
+    "lms/views/feature_flags_test.py",
```

Comment:
Running `make template` removes the lines from `pshell.py` onwards.

Reply:
This is a weird one:

* `make template` does not make any changes to `pshell.py` for me even if I do `make clean template`
* But if I clone another copy of the repo, switch to this branch, and run `make template` then it _does_ make the changes to `pshell.py` :confused:

This is the relevant part of the cookiecutter's `pyproject.toml` template:

https://github.com/hypothesis/cookiecutters/blob/d5ae86cc5817e86a51f689152e0dd49c76c22103/_shared/project/pyproject.toml#L57-L65

The lines that are being removed need to go in a `.cookiecutter/includes/coverage/omit` file. And indeed there is no such `omit` file on this branch, so I would expect the lines to be getting removed by `make template`. So why are these lines not being removed when I run `make template` in my original clone of this repo?

It turns out that I do have a `cookiecutter/includes/coverage/omit` file locally but this file is ignored by one of the cookiecutter's `.gitignore` rules:

https://github.com/hypothesis/cookiecutters/blob/d5ae86cc5817e86a51f689152e0dd49c76c22103/_shared/project/.gitignore#L2

I'll open an issue in the cookiecutter's repo to fix that rule. In the meantime I've force-added the missing file on this branch (even though it's `.gitignore`'d) so `make template` is clean now

Reply:
https://github.com/hypothesis/cookiecutters/issues/123